src/features/subscription/NotificationsConnectionModal.native.test.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+
+import { navigate } from '__mocks__/@react-navigation/native'
+import { NotificationsConnectionModal } from 'features/subscription/NotificationsConnectionModal'
+import { analytics } from 'libs/analytics'
+import { reactQueryProviderHOC } from 'tests/reactQueryProviderHOC'
+import { fireEvent, render, screen, waitFor } from 'tests/utils'
+
+jest.mock('features/profile/pages/NotificationSettings/usePushPermission', () => ({
+  usePushPermission: jest.fn(() => ({
+    pushPermission: 'granted',
+  })),
+}))
+
+const mockDismissModal = jest.fn()
+
+describe('<NotificationsConnectionModal />', () => {
+  it('should render correctly', () => {
+    renderModal(true)
+
+    expect(screen).toMatchSnapshot()
+  })
+
+  it('should dismiss modal on press rightIconButton', () => {
+    renderModal(true)
+
+    const dismissModalButton = screen.getByTestId('rightIcon')
+
+    fireEvent.press(dismissModalButton)
+
+    expect(mockDismissModal).toHaveBeenCalledTimes(1)
+  })
+
+  it('should dismiss modal on press "Créer un compte"', () => {
+    renderModal(true)
+
+    const authButton = screen.getByText('Créer un compte')
+
+    fireEvent.press(authButton)
+
+    expect(mockDismissModal).toHaveBeenCalledTimes(1)
+  })
+
+  it('should go to sign up page on press "Créer un compte"', async () => {
+    renderModal(true)
+
+    const authButton = screen.getByText('Créer un compte')
+
+    fireEvent.press(authButton)
+
+    await waitFor(() => {
+      expect(navigate).toHaveBeenCalledWith('SignupForm', { from: 'home' })
+    })
```

Comment:
C'est plus logique d'utiliser `act` quand on fait une action pour pas avoir de **act warning**. `waitFor` devrait utiliser en dernier recours si jamais le problème persiste
```suggestion
    await act(() => fireEvent.press(authButton))

    expect(navigate).toHaveBeenCalledWith('SignupForm', { from: 'home' })
```